packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -120,7 +120,12 @@ export class Outbox {
 	}
 
 	public get messageCount(): number {
-		return this.attachFlowBatch.length + this.mainBatch.length + this.blobAttachBatch.length;
+		return (
+			this.attachFlowBatch.length +
+			this.mainBatch.length +
+			this.blobAttachBatch.length +
+			this.idAllocationBatch.length
```

Comment:
☹️ This separate queue logic needs to be either made more explicit and self-contained to avoid these types of bugs or be removed altogether. The original idea was to help with the batch size, but now, due to the fact that it adds the guarantees about the order in which the ops are flushed, it became more than that. I think we should explore ways of getting rid of this to pay back the tech debt..

Reply:
No objection here, but I do not have cycles to look into it - maybe open a task?